google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionImpl.java

```diff
@@ -435,14 +466,18 @@ public AsyncTransactionManagerImpl transactionManagerAsync(TransactionOption...
 
   @Override
   public ApiFuture<Empty> asyncClose() {
-    return spanner.getRpc().asyncDeleteSession(getName(), getOptions());
+    XGoogSpannerRequestId reqId =
+        this.getRequestIdCreator().nextRequestId(1 /* TODO: channelId */, 0);
+    return spanner.getRpc().asyncDeleteSession(getName(), reqId.withOptions(getOptions()));
   }
 
   @Override
   public void close() {
     ISpan span = tracer.spanBuilder(SpannerImpl.DELETE_SESSION);
     try (IScope s = tracer.withSpan(span)) {
-      spanner.getRpc().deleteSession(getName(), getOptions());
+      XGoogSpannerRequestId reqId =
+          this.getRequestIdCreator().nextRequestId(1 /* TODO: channelId */, 0);
```

Comment:
```suggestion
          this.getRequestIdCreator().nextRequestId(1 /* TODO: channelId */, 1);
```

Reply:
I manually have been increasing attempts for it. Let's get this merged in as is and in the next round I shall get them all fixed up.